libtensorinterface/include/dpctl_libtensor_linear_sequences_interface.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "ExternC.h"
+#include "dpctl_sycl_types.h"
+
+DPCTL_C_EXTERN_C_BEGIN
+
+typedef struct DPCTLSyclEventPair {
+    DPCTLSyclEventRef first;
+    DPCTLSyclEventRef second;
+} DPCTLSyclEventPair;
+
+DPCTLSyclEventPair 
```

Comment:
Quick comment based on a very superficial review of the changes.

I assume the plan is to call this function from LLVM IR inside a dpex-generated module? If so, returning a `struct` is a not recommended at all. There is a potential to run into all sort of platform-specific ABI issues. Clang on x86 will convert the type signature to return void and add a return value pointer. It might be different on other ABI.

Moreover, I am against creating these C wrappers for C++ `std` classes. I did that for `std::vector` in libsyclinterface to adhere to a pure C API, as we were not sure how the API will be used. If I were writing `dpctl` today I see no need to do it that way as the only place the `std::vector` wrapper is used is in the Cython interface and a better approach will be to make the Cython interface pure C++.

Reply:
What would be a better alternative? Should I return an opaque pointer? Or just keep it `void`? Becuase I don't think we are using that returned pair of events anywhere anyway. They are just used to check if those events were finished or not.

Reply:
We should have the two event objects created by caller and passed to the function as pointers. They will be assigned in the function.